test/common/quic/active_quic_listener_test.cc

```diff
@@ -553,6 +557,23 @@ TEST_P(ActiveQuicListenerTest, QuicRejectsAllAndResumes) {
   EXPECT_TRUE(ActiveQuicListenerPeer::enabled(*quic_listener_));
 }
 
+TEST_P(ActiveQuicListenerTest, EcnReportingIsEnabled) {
+  initialize();
+  Network::Socket *socket = ActiveQuicListenerPeer::socket(*quic_listener_);
+  absl::optional<Network::Address::IpVersion> version = socket->ipVersion();
+  EXPECT_TRUE(version.has_value());
+  int optval;
+  socklen_t optlen = sizeof(optval);
+  Api::SysCallIntResult rv;
+  if (*version == Network::Address::IpVersion::v6) {
+    rv = socket->getSocketOption(IPPROTO_IPV6, IPV6_RECVTCLASS, &optval, &optlen);
+  } else {
+    rv = socket->getSocketOption(IPPROTO_IP, IP_RECVTOS, &optval, &optlen);
```

Comment:
Does this work on Windows?

Reply:
I haven't tested it (as we discussed) but that's the point of the change to os_sys_calls_impl.